phaser/steps.py

```diff
@@ -127,12 +127,12 @@ def postprocess(step_function, result):
         if not isinstance(result, Sequence):
             raise PhaserError(
                 f"Step {step_function} returned a {result.__class__} rather than a list of rows")
-        return result
+        return result, check_size
```

Comment:
The phase passes a `context` into the step function call, which translates to a call to `_step_wrapper`.  In that function, if you pass `context` into `self.postprocess`, then you can do the checks for row counts in this `postprocess` function.  That is cleaner than passing back a tuple, but unfortunately, you would need to duplicate the size checks in both `batch_step` and `dataframe_step`.

Reply:
Moving this to an issue - maybe come back to it sometime when I'm feeling smart.